### ISSUE TITLE
[Bug] Paste an URL in Board failed - OB 1.7.2 - On Windows 10

### ISSUE THREAD

**Issue by sebojolais** [OpenBoard-org/OpenBoard#1184]
### Describe the bug
When you paste an URL in the Board mode, the custom application is not shown and an error message is shown shortly at the bottom left.

### To Reproduce

1. Open OB
2. Got to the Board view
3. On an empty page
4. Copy any kind of URL from the WEB
5. Paste it on the empty page in the Board view of OB

#### Expected behavior
A custom application with the page of the URL should be visible.

#### Actual behavior
No application is built and an error message is shown at the bottom left, but it is shown too quickly, I was not able to read it.

### Context
  - reproducibility : `systematic` 
  - OpenBoard version :
      - bug happens with : `1.7.2`
      - bug doesn't happen with : `1.7.0` `1.7.1`
  - OS : 
      - bug happens with : `Windows 10`
      - bug doesn't happen with :`Linux`
      - didn't check :`macOS` 
  - Installation source : `OpenBoard download page`

**Comment by letsfindaway** [comment]
I just tried that e.g. 

- with the URL of this issue and it was working fine for me. I got the dialog asking whether to create an application for that URL. 
- I also tried it with the URL of an image, and the image was downloaded and inserted on the board. 
- With the URL to a `.tgz` archive I got an error message "Unknown content" in the lower left corner.

All of that is expected for me. So do you experience this problem actually with **all** kinds of URLs pointing to **any** type of content?

**Comment by sebojolais** [comment]
Hi @letsfindaway ,
Thank you for your quick answer.
All of that is expected for me too on linux.
But did you test also on Windows ?

**Comment by letsfindaway** [comment]
> But did you test also on Windows ?

No, don't have Windows. But could you make a screen video to catch the error message?
This https://www.microsoft.com/en-us/windows/learning-center/how-to-record-screen-windows-11 should also work on Windows 10.

**Comment by sebojolais** [comment]
Below is the error message when I paste the URL : https://primaire.forge.apps.education.fr/
![2024-12-12 15_50_28-OB_Win10_PasteURL avi - Lecteur multimédia VLC](https://github.com/user-attachments/assets/81e36a67-26ec-488a-981c-f8d01ce27c6b)

**Comment by letsfindaway** [comment]
That's a rather generic error message. But we at least know now that it is emitted at `UBBoardController::1198`, because `downloadFinished` was called with `pSuccess = false`.

After digging deeper, this is most probably initiated here:

https://github.com/OpenBoard-org/OpenBoard/blob/80fb7dd6e52bf2a2e2cf236f77f6628c0c12ec14/src/network/UBHttpGet.cpp#L97-L106

Here a warning would be logged with some more information. So could you either try to start OpenBoard in a console with the `-log` flag? This should create a log file in a `/log` subdirectory of your OpenBoard user data directory. On Windows it should be `%USERPROFILE%\AppData\Local\OpenBoard\log`.

Could you please check that whether this helps us locating this problem?

**Comment by sebojolais** [comment]
The log is:
```
2024-12-12T22:22:57.747      Loaded Qt Gui translations
2024-12-12T22:22:57.749      QtWebEngine translations in  "fr"  are not available or could not be loaded
2024-12-12T22:22:57.749      Running application in: "fr"
2024-12-12T22:22:58.028      file name argument ""
2024-12-12T22:22:58.028      availableScreens QList(QScreen(0x22134286650, name="\\\\.\\DISPLAY1"))
2024-12-12T22:22:58.600      Injecting qwebchannel.js
2024-12-12T22:22:58.651      Injecting qwebchannel.js
2024-12-12T22:22:58.731      userDocumentDirectory() "C:/Users/sebastien/AppData/Local/OpenBoard/document"
2024-12-12T22:22:58.732      userDocumentDirectory() "C:/Users/sebastien/AppData/Local/OpenBoard/document"
2024-12-12T22:22:58.863      process starts
2024-12-12T22:22:58.864      userDocumentDirectory() "C:/Users/sebastien/AppData/Local/OpenBoard/document"
2024-12-12T22:22:58.885      loadDocumentScene: got result from cache
2024-12-12T22:22:58.923      Private browsing false
2024-12-12T22:22:58.924      Using cookie policy "DenyThirdParty"
2024-12-12T22:22:58.995      assignRoles using screen list QList()
2024-12-12T22:22:58.996      positionScreens
2024-12-12T22:22:59.012      control geometry QRect(0,0 1920x1080)
2024-12-12T22:22:59.958      Checking for update at url:  "http://www.openboard.ch/update.json"
2024-12-12T22:23:00.627      The backend "cert-only" does not support QSslKey
2024-12-12T22:23:00.628      Active TLS backend does not support key creation
2024-12-12T22:23:00.628      The backend "cert-only" does not support QSslKey
2024-12-12T22:23:00.628      Active TLS backend does not support key creation
2024-12-12T22:23:00.629      The backend "cert-only" does not support QSslKey
2024-12-12T22:23:00.629      Active TLS backend does not support key creation
2024-12-12T22:23:00.629      The backend "cert-only" does not support QSslKey
2024-12-12T22:23:00.629      Active TLS backend does not support key creation
2024-12-12T22:23:00.724      The backend "cert-only" does not support QSslKey
2024-12-12T22:23:00.725      Active TLS backend does not support key creation
2024-12-12T22:23:00.725      The backend "cert-only" does not support QSslKey
2024-12-12T22:23:00.726      Active TLS backend does not support key creation
2024-12-12T22:23:00.727      The backend "cert-only" does not support QSslSocket
2024-12-12T22:23:00.727      The backend named "cert-only" does not support TLS
2024-12-12T22:23:00.728      QSslSocket::connectToHostEncrypted: TLS initialization failed
2024-12-12T22:23:00.730      Error downloading update file:  "TLS initialization failed"
2024-12-12T22:23:28.680      something has been dropped on the board! Url is:  "https://primaire.forge.apps.education.fr/"
2024-12-12T22:23:28.717      Actual downloads size:  1
2024-12-12T22:23:28.719      Actual downloads size:  1
2024-12-12T22:23:29.781      The backend "cert-only" does not support QSslKey
2024-12-12T22:23:29.782      Active TLS backend does not support key creation
2024-12-12T22:23:29.783      The backend "cert-only" does not support QSslKey
2024-12-12T22:23:29.785      Active TLS backend does not support key creation
2024-12-12T22:23:29.786      The backend "cert-only" does not support QSslSocket
2024-12-12T22:23:29.787      The backend named "cert-only" does not support TLS
2024-12-12T22:23:29.788      QSslSocket::connectToHostEncrypted: TLS initialization failed
2024-12-12T22:23:29.790      "https://primaire.forge.apps.education.fr/" get finished with error :  QNetworkReply::UnknownNetworkError
2024-12-12T22:23:29.815      Actual downloads size:  0
2024-12-12T22:23:29.816      Actual downloads size:  0
2024-12-12T22:23:34.389      Persisting document metadata; path is "C:/Users/sebastien/AppData/Local/OpenBoard/document/OpenBoard Document 2024-12-12 22-22-58.864/metadata.rdf"
2024-12-12T22:23:34.389      User settings saved
2024-12-12T22:23:34.398      User settings saved
2024-12-12T22:23:34.402      application is quitting
2024-12-12T22:23:34.472      application Will close signal received
2024-12-12T22:23:34.473      start waiting
2024-12-12T22:23:34.475      stop waiting after  2  ms
```
- Where I started OB at 22h 23min
- I pasted the URL at 22h 23min and 28s
- And I closed OB at 22h 23min and 34s

**Comment by letsfindaway** [comment]
According to https://www.qt.io/blog/moving-to-openssl-3-in-binary-builds-starting-from-qt-6.5-beta-2 we have to switch to OpenSSL 3 when we're using Qt 6.5 or higher. On Linux we typically don't have this problem as people already have OpenSSL 3 on their machines. 

AFAIK the windows version is using Qt 6.7.2, but still bundles OpenSSL 1.1. This seems to be the source of this problem. @kaamui: I think this has to be resolved in the Windows release scripts.

Note that the log shows that even the access to the version check does not work, as it is also a TLS connection. The `http://www.openboard.ch/update.json` URL is redirected to a `https:` connection by the server.

This might also affect the MacOS version, as I see OpenSSL 1.1 in https://github.com/OpenBoard-org/OpenBoard/blob/master/release_scripts/osx/relink_libs.sh.

Thank you very much for pointing this out!

**Comment by kaamui** [comment]
Oops I missed that new requirement.

> This might also affect the MacOS version, as I see OpenSSL 1.1 in https://github.com/OpenBoard-org/OpenBoard/blob/master/release_scripts/osx/relink_libs.sh.

this script is no longer used, but I'll check if the issue occurs on macOS too. 

Thank you for the help @sebojolais and @letsfindaway  !

**Comment by kaamui** [comment]
I just looked at my Windows machine and I actually have openssl 3.0.15 on my OpenBoard third-party repository, locally. 

I now remember that I had retained this new version to 3.0.15 because of a major security issue only occurring in 3.1, and probably read at the time it was more reasonable to use 3.0 versions.

**Comment by kaamui** [comment]
@sebojolais I pushed the folder on the third parties. Until a new release is available, you can just copy paste the `libcrypto-3-x64.dll` and `libssl-3-x64.dll` to the installation folder to fix the issue

**Comment by sebojolais** [comment]
@kaamui I never use Windows myself, but I heard some complains from colleagues. I am happy if this ticket is helpful.

> I pushed the folder on the third parties (dev branch). Until a new release is available, you can just copy paste the libcrypto-3-x64.dll and libssl-3-x64.dll to the installation folder to fix the issue

I think you mean this repository : https://github.com/OpenBoard-org/OpenBoard-ThirdParty/tree/dev/openssl/openssl-3.0.15-win64/bin
Is not it ?

**Comment by kaamui** [comment]
Yep !

**Comment by kaamui** [comment]
Should be fixed in 1.7.3
